File: test_data/code/python/1128-number-of-equivalent-domino-pairs.py
class Solution:
    def numEquivDominoPairs(self, dominoes: List[List[int]]) -> int:
        count = {}
        for domino in dominoes:
            domino.sort()  # sort the domino to make it unique
            count[tuple(domino)] = count.get(tuple(domino), 0) + 1
        pairs = 0
        for v in count.values():
            pairs += v * (v - 1) // 2  # calculate the number of pairs for each unique domino
        return pairs

File: test_data/code/java/1128-number-of-equivalent-domino-pairs.java
class Solution {
    public int numEquivDominoPairs(int[][] dominoes) {
        HashMap<String, Integer> map = new HashMap<>();
        for (int i = 0; i < dominoes.length; i++) {
            String key = Integer.toString(dominoes[i][0]) + "," + Integer.toString(dominoes[i][1]);
            map.put(key, map.getOrDefault(key, 0) + 1);
        }
        int res = 0;
        for (int count : map.values()) {
            res += count * (count - 1) / 2;
        }
        return res;
    }
}